src/main/java/com/lgtm/easymoney/services/impl/RequestServiceImpl.java

```diff
@@ -96,136 +93,67 @@ public Transaction createRequest(User reqBy, User reqTo, BigDecimal amount, Stri
   }
 
   @Override
-  public ResponseEntity<RequestRsp> createRequest(RequestReq req) {
-    // get params
-    Long fromUid = req.getFromUid();      // from = req by
-    Long toUid = req.getToUid();          // to = req to
-    BigDecimal amount = req.getAmount();
-    Category category = req.getCategory();
-    String desc = req.getDescription();
+  public ResourceCreatedRsp createRequest(RequestReq req) {
     // create a request
-    User fromUser = userService.getUserById(fromUid);
-    User toUser = userService.getUserById(toUid);
-    Transaction trans = createRequest(fromUser, toUser, amount, desc, category);
-    // payload
-    // todo need to refactor some of these
-    RequestRsp res = new RequestRsp();
-    res.setSuccess(trans != null);
-    res.setCurrBalance(fromUser.getBalance());
-    res.setRequests(transactionService.generateListResponseFromTransactions(
-            new ArrayList<>(Arrays.asList(trans))));
-    res.setMessage("Request has been created successfully!");
-    if (trans != null) {
-      return ResponseEntity.status(HttpStatus.OK).body(res);
-    }
-    // TODO refactor this?
-    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(res);
+    Transaction trans = createRequest(
```

Comment:
When would this return null? If there's problems creating a new one, a DB exception should be thrown